mono/metadata/threadpool-ms.c

```diff
@@ -686,29 +695,54 @@ worker_try_create (void)
 {
 	ThreadPoolCounter counter;
 	MonoInternalThread *thread;
+	gint32 now;
+
+	mono_mutex_lock (&threadpool->worker_creation_lock);
```

Comment:
This is really bad coding.

It's calling mono_thread_create_internal with this lock held. The lock is not a leaf lock
and that function takes a ton of locks itself. This break the allowed runtime locking rules.

How frequently is worker_try_create called during steady state?

What's the perf impact of this change?

Why 10 threads per second?